auth-api/src/auth_api/services/api_gateway.py

```diff
@@ -0,0 +1,126 @@
+            create_consumer_payload = dict(email=email,
+                                           firstName=org.name,
+                                           lastName=org.branch_name or 'BCR',
+                                           userName=org.name,
+                                           clientId=client_rep.get('clientId'),
+                                           clientSecret=client_rep.get('secret'),
+                                           apiAccess=['ALL_API'],
+                                           apiKeyName=name)
+            api_key_response = RestService.post(
+                f'{consumer_endpoint}/mc/v1/consumers',
+                additional_headers={'x-apikey': gw_api_key},
+                data=create_consumer_payload,
+                generate_token=False
+            )
+            org.has_api_access = True
+            org.save()
+        else:
+            # Create additional API Key if a consumer exists
+            api_key_response = RestService.post(
+                f'{consumer_endpoint}/mc/v1/consumers/{email}/apikeys',
+                additional_headers={'x-apikey': gw_api_key},
+                data=dict(
+                    apiAccess=['ALL_API'],
+                    apiKeyName=name
+                ),
+                generate_token=False
+            )
+
+        return api_key_response.json()
+
+    @classmethod
+    def revoke_key(cls, org_id: int, api_key: str):
+        """Revoke api key."""
+        current_app.logger.debug('<revoke_key ')
+        consumer_endpoint: str = current_app.config.get('API_GW_CONSUMERS_API_URL')
+        gw_api_key: str = current_app.config.get('API_GW_KEY')
+        email_id = cls._get_email_id(org_id)
+
+        RestService.patch(
+            f'{consumer_endpoint}/mc/v1/consumers/{email_id}/apiKeys/{api_key}?action=removeAccess',
+            additional_headers={'x-apikey': gw_api_key},
+            data=dict(apiAccess='ALL_API'),
+            generate_token=False
+        )
+
+    @classmethod
+    def get_api_keys(cls, org_id: int) -> List[Dict[str, any]]:
+        """Get all api keys."""
+        current_app.logger.debug('<get_api_keys ')
+        consumer_endpoint: str = current_app.config.get('API_GW_CONSUMERS_API_URL')
+        gw_api_key: str = current_app.config.get('API_GW_KEY')
+
+        email_id = cls._get_email_id(org_id)
+        try:
+            consumers_response = RestService.get(
+                f'{consumer_endpoint}/mc/v1/consumers/{email_id}',
+                additional_headers={'x-apikey': gw_api_key}
+            )
+            api_keys = consumers_response.json()
+        except HTTPError as exc:
+            if exc.response.status_code == 404:
+                api_keys = {}
+            else:
+                raise exc
+
+        return api_keys
+
+    @classmethod
+    def _get_email_id(cls, org_id) -> str:
+        if current_app.config.get('API_GW_CONSUMER_EMAIL', None):
```

Comment:
is it if NOT none?

Reply:
This works, but will change to be consistent :+1